File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SmartphoneNode : public rclcpp::Node {

    public:
        SmartphoneNode() : Node("smartphone"){

           // Initialize the suscriber
           suscriber_ = this->create_subscription<example_interfaces::msg::String>("robots_news", 
                                                                            10, 
                                                                            std::bind(&SmartphoneNode::callbackRobotNews, this,
                                                                            std::placeholders::_1));
            RCLCPP_INFO(this->get_logger(), "Smartphone has been started"); 
        }
        
    private:
        // Callback function
        void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg){

            RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
        }

        // Making the suscriber
        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr suscriber_;
};

int main(int argc, char**argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.msg import String


class SmarthponeNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        
        # Making a suscriber (type, name_of_topic, callback_function, buffer)
        self.suscriber_ = self.create_subscription(String, "robots_news", 
                                                   self.callback_robot_news, 10)
        
        self.get_logger().info("Smartphone has been started.")

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = SmarthponeNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()